### DISCUSSION TITLE
How does DBeaver generate the database hierarchy folder structure?

### DISCUSSION THREAD

**Discussion by GAZ082** [dbeaver/dbeaver#8882] [Q&A] [answered]
I guess I did a faulty database restoration or the dump did not have this data and the "folder like" structure was lost. This is a capture of Dbeaver showing what I mean with "folder like" structure. I can easily access the functions, tables, schemas, etc. stored in the database definition. I'm not so worried about the loss of the original structure as it did not have much, but I do want to recreate it step by step when I need it. For example, now I made a function and would like to see it under a "Functions" folder, how can I do that?

Thanks.

![image](https://user-images.githubusercontent.com/100194/83648836-e6898c80-a58c-11ea-8ca8-fb1e45f49596.png)

**Comment by uslss** [comment, marked as answer]
The structure depends on the driver, not the database.  So, e.g., the function should appear in the folder after its creation.
